### PR TITLE
WildFly datasources v2

### PR DESCRIPTION
**Pull request by jfdenise**
* More configuration points for all datasources (ENV variables and System properties). 
* Add MSSQLServer
* Add Mariadb
* Documentation rework
* Add testing based on WildFly bootable JAR.

### CODE REVIEW

File: galleon-feature-pack/src/main/resources/modules/com/microsoft/sqlserver/jre8/main/module.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module name="com.microsoft.sqlserver.jre8" xmlns="urn:jboss:module:1.8">
```
Comment (jmesnil) [inline galleon-feature-pack/src/main/resources/modules/com/microsoft/sqlserver/jre8/main/module.xml:2]: should to be simply named `com.microsoft.sqlserver`? (jre8 seems to be an implementation detail for the version only)

Comment (jmesnil) [reply galleon-feature-pack/src/main/resources/modules/com/microsoft/sqlserver/jre8/main/module.xml:2]: and it would be consistent with Oracle naming

**Review by jmesnil — CHANGES_REQUESTED** [review]
aside from my question about the name of the JBoss Module for MQ SQL Server, this looks good!

File: galleon-feature-pack/src/main/resources/modules/com/microsoft/sqlserver/jre8/main/module.xml (hunk 1 above)
Comment (jfdenise) [reply galleon-feature-pack/src/main/resources/modules/com/microsoft/sqlserver/jre8/main/module.xml]: @jmesnil I reworked the mssqlserver driver name.

File: galleon-feature-pack/pom.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.wildfly</groupId>
+        <artifactId>wildfly-datasources-galleon-pack-parent</artifactId>
+        <version>1.2.4.Final-SNAPSHOT</version>
```
Comment (jmesnil) [inline galleon-feature-pack/pom.xml:23]: we can do that after this PR is merged but let's bump to 2.0.0.Final-SNAPSHOT for the next release as we have made major changes (including new DBs)

Comment (jfdenise) [reply galleon-feature-pack/pom.xml]: @jmesnil, PR is ready.

**Review by jmesnil — APPROVED** [review]

**Comment by jmesnil** [comment]
@jfdenise thanks! that's a great new v2 :)

**Pull request merged**